src/user/repository/users-repository.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@nestjs/common';
+import { AutoSuggestUserInfoDto } from '../dto/autoSuggestUserInfo.dto';
+import { User } from '../models/user.model';
+import { InjectModel } from '@nestjs/sequelize';
+import { Op } from 'sequelize';
+import { NewUserDto } from '../dto/new-user.dto';
+import { UserDto } from '../dto/user.dto';
+import { UpdateUserDto } from '../dto/update-user.dto';
+
+@Injectable()
+export class UserRepository {
+
+  constructor(@InjectModel(User) private userModel: typeof User) {}
+
+  async getAll(): Promise<User[]> {
+    return this.userModel.findAll({ where: { isDeleted: false }});
+  }
+
+  async getOne(id: string): Promise<User | null> {
+    return this.userModel.findOne({ where: { [Op.and]: [{ id }, { isDeleted: false }] } });
+  }
+
+  async getSome(params: AutoSuggestUserInfoDto): Promise<UserDto[]> {
+    return this.userModel.findAll({
+      where: { login: {[Op.iLike]: `%${params.loginSubstring}%` }},
+      limit: params.limit
+    });
+  }
+
+  async create(newUserInfo: NewUserDto): Promise<UserDto | null> {
+    const isLoginUnique = await this.checkIsLoginUnique(newUserInfo.login);
+    return isLoginUnique ? await this.userModel.create(newUserInfo) : null;
+  }
+
+  async remove(id: string): Promise<User | null> {
+    await this.userModel.update({ isDeleted: true }, { where: { id } });
+    return this.userModel.findOne({ where: { id } });
```

Comment:
may be you do not need to return user here?

Reply:
changed the method

Reply:
> * in case of invalid id service return status code 500 “Internal server error”, I think there should be 404
> * to avoid code duplication in user dtos I suggest to use “extends”
>   `export class UpdateUserDto extends NewUserDto {}`
> * get all users with "loginSubstring" returns users, that were previously deleted (with prop isDeleted: true)

1. I completely agree with you and I hope i fixed all the scenarios when the mistake appeared
2. The differance between dtos are error descriptions. Ofcorse I can rewrite them but I would like to save.
3. added check if the user is deleted and sort by login.